### DISCUSSION TITLE
Python Connection Thread Affinity

### DISCUSSION THREAD

**Discussion by celsius-233** [duckdb/duckdb#2990] [Q&A]
Hi,

For the Python API, does a duckdb connection have thread affinity, i.e. can I create a connection on one thread and execute a query on another using that connection instance?  Thanks in advance.

-D

**Comment by blackrez** [comment]
Hello,

You should use cursor from connect it will provide thread support.

I write an example here.

https://gist.github.com/blackrez/d1216dfd8b7f062f071a7bcd4596469f
